### PR TITLE
Verify MSAL-CPP: Update macOS/Xcode versions to match MSAL-CPP

### PR DESCRIPTION
**Pull request by remyjette**
## Type of change

- [ ] Feature work
- [ ] Bug fix
- [ ] Documentation
- [X] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [X] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

**Comment by juan-arias** [comment]
@remyjette FYI, the `[MSAL C++] Common core subtree update check iOS` check will fail until we finish the 1.7.36 release https://github.com/AzureAD/microsoft-authentication-library-common-for-objc/pull/1379

**Comment by remyjette** [comment]
@juan-arias Yeah, I know. I just wanted to also get rid of this warning:
> The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 13.3, but the range of supported deployment target versions is 10.13 to 13.0.99. (in target 'MSALSampleApp' from project 'msalsdk')

File: azure_pipelines/verify_msalcpp_per_pr_mac.yml [hunk 1]
```diff
@@ -49,9 +49,9 @@ jobs:
      arguments: '--skip-checkout --msal-common-repo $(Agent.BuildDirectory)/CommonCore --remote-ref $(commit_number)'
 
   - task: UsePythonVersion@0
-    displayName: 'Switch to use Xcode 14.1'
+    displayName: 'Switch to use Xcode 15.3'
```
Comment (Veena11) [inline azure_pipelines/verify_msalcpp_per_pr_mac.yml:52]: Latest macOS14 agent supports xcode 15.4, so in broker we are updating to use 15.4. would you be able to keep it the same ?

Comment (Veena11) [reply azure_pipelines/verify_msalcpp_per_pr_mac.yml:52]: Broker PR for your reference : https://github.com/AzureAD/azure-activedirectory-tokenbroker-for-objc/pull/1646

**Comment by remyjette** [comment]
Currently MSAL-C++ is still using the macOS-13 agent for building iOS, which also has a maximum Xcode version of 15.2

Petar made a PR to update it https://github.com/AzureAD/microsoft-authentication-library-for-cpp/pull/4264, but looks like that missed some jobs and also didn't change the Xcode version

I've bumped them in https://github.com/AzureAD/microsoft-authentication-library-for-cpp/pull/4289, if that succeeds then I will merge it and update this PR to match

File: azure_pipelines/verify_msalcpp_per_pr_mac.yml (hunk 1 above)
Comment (remyjette) [reply azure_pipelines/verify_msalcpp_per_pr_mac.yml:52]: See comment below

**Review by kaisong1990 — APPROVED** [review]

File: azure_pipelines/verify_msalcpp_per_pr_mac.yml (hunk 1 above)
Comment (remyjette) [reply azure_pipelines/verify_msalcpp_per_pr_mac.yml:52]: Updated

**Comment by remyjette** [comment]
Whoops, deleted the wrong branch

Anyway, updated to macOS-14 and Xcode 15.4 for both macOS and iOS

File: azure_pipelines/verify_msalcpp_per_pr_ios.yml [hunk 2]
```diff
@@ -48,12 +48,12 @@ jobs:
...
   - script: |
-     /bin/bash -c "sudo xcode-select -s /Applications/Xcode_15.2.app"
+     /bin/bash -c "sudo xcode-select -s /Applications/Xcode_15.4.app"
 
   - template: azure_pipelines/templates/ios-setup.yml@microsoft-authentication-library-for-cpp
-  
+
   - task: PythonScript@0
     name: Build
     displayName: 'Build x64 Debug iOS'
```
Comment (Veena11) [inline azure_pipelines/verify_msalcpp_per_pr_ios.yml:59]: would this need change in build.py to use the correct ios versions as well ? iOS15 and Simulator 17.5 ?

Comment (remyjette) [reply azure_pipelines/verify_msalcpp_per_pr_ios.yml:59]: This is calling MSAL-CPP's build.py, so changes to msal-common-for-objc's build.py are out of scope for this PR

Comment (Veena11) [reply azure_pipelines/verify_msalcpp_per_pr_ios.yml:59]: aah ok, got it

**Review by Veena11 — APPROVED** [review]

**Pull request merged**